File: mobile/src/main/java/treehou/se/habit/ui/sitemaps/sitemaplist/SitemapListComponent.java
package treehou.se.habit.ui.sitemaps.sitemaplist;


import dagger.Subcomponent;
import treehou.se.habit.module.ActivityScope;
import treehou.se.habit.module.FragmentComponent;
import treehou.se.habit.module.FragmentComponentBuilder;

@ActivityScope
@Subcomponent(
        modules = SitemapListModule.class
)
public interface SitemapListComponent extends FragmentComponent<SitemapListFragment> {

    @Subcomponent.Builder
    interface Builder extends FragmentComponentBuilder<SitemapListModule, SitemapListComponent> {

    }
}

File: mobile/src/main/java/treehou/se/habit/ui/sitemaps/sitemaplist/SitemapListComponent.kt
package treehou.se.habit.ui.sitemaps.sitemaplist


import dagger.Subcomponent
import treehou.se.habit.module.ActivityScope
import treehou.se.habit.module.FragmentComponent
import treehou.se.habit.module.FragmentComponentBuilder

@ActivityScope
@Subcomponent(modules = arrayOf(SitemapListModule::class))
interface SitemapListComponent : FragmentComponent<SitemapListFragment> {

    @Subcomponent.Builder
    interface Builder : FragmentComponentBuilder<SitemapListModule, SitemapListComponent>
}